s2a/src/main/java/io/grpc/s2a/handshaker/S2ATrustManager.java

```diff
@@ -121,6 +121,7 @@ private void checkPeerTrusted(X509Certificate[] chain, boolean isCheckingClientC
     try {
       resp = stub.send(reqBuilder.build());
     } catch (IOException | InterruptedException e) {
```

Comment:
This catch needs to be split, because we don't want to interrupt on IOException.

Reply:
Thanks for pointing that out, done in https://github.com/grpc/grpc-java/pull/11534/commits/b060a4929d9d172ec287d28f50d750b502ddc8f5